pkg/alertmanager/alertmanager_ring.go

```diff
@@ -42,17 +42,20 @@ var SyncRingOp = ring.NewOp([]ring.InstanceState{ring.ACTIVE, ring.JOINING}, fun
 // is used to strip down the config to the minimum, and avoid confusion
 // to the user.
 type RingConfig struct {
-	KVStore           kv.Config     `yaml:"kvstore" doc:"description=The key-value store used to share the hash ring across multiple instances."`
-	HeartbeatPeriod   time.Duration `yaml:"heartbeat_period"`
-	HeartbeatTimeout  time.Duration `yaml:"heartbeat_timeout"`
-	ReplicationFactor int           `yaml:"replication_factor"`
+	KVStore              kv.Config     `yaml:"kvstore" doc:"description=The key-value store used to share the hash ring across multiple instances."`
+	HeartbeatPeriod      time.Duration `yaml:"heartbeat_period"`
+	HeartbeatTimeout     time.Duration `yaml:"heartbeat_timeout"`
+	ReplicationFactor    int           `yaml:"replication_factor"`
+	ZoneAwarenessEnabled bool          `yaml:"zone_awareness_enabled"`
```

Comment:
With the zone-awareness, these fields match [ring.Config](https://github.com/cortexproject/cortex/blob/master/pkg/ring/ring.go#L126) now. It would be a nice bit of cleanup to use it here IMO:

```suggestion
	RingConfig      ring.Config `yaml:",inline"`
	HeartbeatPeriod time.Duration `yaml:"heartbeat_period"`
```

Reply:
Hum.. I can definitely do that, but i was more following the pattern we used on the [gateway ring](https://github.com/cortexproject/cortex/blob/master/pkg/storegateway/gateway_ring.go#L52).

Doing this change will make we use those properties like `cfg.ShardingRing.RingConfig.ZoneAwarenessEnabled`, what do you think?

Reply:
Ah I see... that must be where this code came from. I think we can do the refactoring so that the config file and command line don't change, but I won't block on this, we can do it in a separate PR.